impress-org/givewp-next-gen
# Feature: Render a selected v(3) form in the block editor

## Description

This PR adds a Block Preview component to the Block Editor when a formId is set. This enables users to see a preview of their selected form. It's rendered using the IframeResizer and the forms URL. 

In order to allow the block to be selected and editable, the iframes css property "pointer events" has been set to none. It is reenabled once the block is selected allowing users to navigate the iframe.

In addition to the above the `useFormOptions` hook was updated to filter the v3 forms by the excerpt.rendered output.
It was originally filtered by the excerpt length, however in v2 forms users can add an excerpt. When added, the v2 form would be included in the filter and display as a form option.

## Affects

Donation form v3

## Visuals

https://www.loom.com/share/6c529105b515466b89fa678d2d542fe3?sid=9e4594bd-f227-4991-ba7e-6e4e4214c911

## Testing Instructions

- Create a v3 form for each form design "classic, multi step, developer."
- add the v3 form block
- Select the forms
- Verify the selected forms render inside the block editor.
-  Verify you can select the block as well as navigate the iframe
- Test switching forms once the block is selected

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204982550616184
